Mage.Sets/src/mage/cards/t/TunnelTipster.java

```diff
@@ -0,0 +1,104 @@
+package mage.cards.t;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfEndStepTriggeredAbility;
+import mage.abilities.condition.Condition;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.abilities.mana.GreenManaAbility;
+import mage.constants.SubType;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.TargetController;
+import mage.constants.WatcherScope;
+import mage.constants.Zone;
+import mage.counters.CounterType;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ZoneChangeEvent;
+import mage.watchers.Watcher;
+
+/**
+ *
+ * @author DominionSpy
+ */
+public final class TunnelTipster extends CardImpl {
+
+    public TunnelTipster(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{G}");
+
+        this.subtype.add(SubType.MOLE);
+        this.subtype.add(SubType.SCOUT);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(1);
+
+        // At the beginning of your end step, if a face-down creature entered the battlefield under your control this turn, put a +1/+1 counter on Tunnel Tipster.
+        this.addAbility(new BeginningOfEndStepTriggeredAbility(
```

Comment:
Try to add ConditionalHint with related conditional and info for a user. It’s a good code style (if you see conditional or amount then card hint can be used too).